### ISSUE TITLE
Off-machine node fails with NO OUTPUT FILE PRODUCED when transcoding

### ISSUE THREAD

**Issue by Fdebijl** [HaveAGitGat/Tdarr#362]
**Describe the bug**
I have a remote gaming server lying around that doesn't see much use, so I decided to put it to work transcoding my library alongside the media server itself. When issuing transcodes this node fails with a warning that no output file was produced. As per the docs I mapped the transcode cache and media share as a network drive on the node and I have verified I can read and write to both. The logs below seem to indicate that the path translation is not being applied, so I would suspect this is a configuration error on my end, or an issue with the path translators.

<details>
<summary>Node Logs</summary>
<p>

```
2.00.031
C:/Users/Parsec/Documents/Tdarr/configs/Tdarr_Node_Config.json
{
  nodeID: 'Parsec',
  nodeIP: '10.10.10.130',
  nodePort: '8267',
  serverIP: '10.10.10.6',
  serverPort: '8266',
  handbrakePath: '',
  ffmpegPath: '',
  mkvpropeditPath: '',
  pathTranslators: [
    { server: 'C:/TransCache/Television', node: 'Y:/Television' },
    { server: 'E:/Television', node: 'X:/Television' }
  ]
}
[2021-02-09T23:58:35.531] [INFO] Tdarr_Node - Starting Tdarr_Node
Preparing environment
[2021-02-09T23:58:35.545] [INFO] Tdarr_Node - Updating plugins
[2021-02-09T23:58:35.546] [INFO] Tdarr_Node - Cloning plugins
[2021-02-09T23:58:35.550] [INFO] Tdarr_Node - Tdarr_Node listening at http://localhost:8267
[2021-02-09T23:58:35.821] [INFO] Tdarr_Node - Finished downloading plugins!
[2021-02-09T23:58:36.920] [INFO] Tdarr_Node - ---------------Binary tests start----------------
[2021-02-09T23:58:36.924] [INFO] Tdarr_Node - handbrakePath:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/HandBrakeCLI.exe
[2021-02-09T23:58:36.927] [INFO] Tdarr_Node - ffmpegPath:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe
[2021-02-09T23:58:36.930] [INFO] Tdarr_Node - mkvpropedit:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/mkvpropedit/windows/mkvpropedit.exe
[2021-02-09T23:58:36.931] [INFO] Tdarr_Node - Binary test 1: handbrakePath working
[2021-02-09T23:58:36.931] [INFO] Tdarr_Node - Binary test 2: ffmpegPath working
[2021-02-09T23:58:36.931] [WARN] Tdarr_Node - Binary test 3: mkvpropeditPath not working:
[2021-02-09T23:58:36.931] [INFO] Tdarr_Node - ---------------Binary tests end-------------------
[2021-02-09T23:58:38.446] [INFO] Tdarr_Node - [2.9s]Plugin update finished
[2021-02-09T23:59:08.002] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCEDC:/TransCache/Television/Woke - S01E05 - Oaktown Get-Down HDTV-1080p-TdarrCacheFile-cnH0JXNXJ.mp4
[2021-02-09T23:59:26.593] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCEDC:/TransCache/Television/The Marvelous Mrs. Maisel - S02E01 - Simone WEBDL-1080p-TdarrCacheFile-M_2UPBbFZ.mp4
[2021-02-09T23:59:38.870] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCEDC:/TransCache/Television/The Marvelous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p-TdarrCacheFile-mXg-kWDoK.mp4
```

</p>
</details>  
 
<details>
<summary>Server Logs</summary>
<p>

```
NodeID: Parsec

Command:

C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/HandBrakeCLI.exe -i E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvelous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p.mkv -o C:/TransCache/Television/The Marvelous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p-TdarrCacheFile-mXg-kWDoK.mp4 -Z General/Fast 1080p30 -e qsv_h264 --all-audio --all-subtitles --optimize

Last 200 lines of CLI log:

[23:59:37] Compile-time hardening features are enabled

[23:59:38] Nvenc version 11.0

[23:59:38] hb_init: starting libhb thread

[23:59:38] thread 1 started ("libhb")

HandBrake 1.3.3 (2

020061300) - MinGW x86_64 - https://handbrake.fr

4 CPUs detected

Opening E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvel

ous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p.mkv...

[23:59:38] CPU: Intel(R) Core(TM) i5-6600K CPU @ 3.50GHz

[23:59:38] - Intel microarchitecture Skylake

[23:59:38] - logical processor count: 4

[23:59:38] Intel Quick Sync Video support: no

[23:59:38] hb_scan: path=E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvelous Mrs. Maisel - S02E08 - Someday. WEBDL-1080p.mkv, title_index=1

src/libbluray/disc/disc.c:323: failed opening UDF image E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvelous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p.mkv

src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv

src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv

src/libbluray/bluray.c:2585: nav_get_title_list(E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvelous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p.mkv\) failed

[23:59:38] bd: not a bd - trying as a stream/file instead

libdvdnav: Using dvdnav version 6.0.1

libdvdread: Encrypted DVD support unavailable.

libdvdread: Could not open input: No such file or directory

libdvdread: Can't

open E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvelous Mrs. Maisel - S02E08 - Someday� WEBDL-1080p.mkv for reading

libdvdnav: vm: failed to open/read the DVD

[23:59:38] dvd: not a dvd - trying as a stream/file instead

[23:59:38] hb_stream_open: open E:/Television/The Marvelous Mrs. Maisel/Season 2/The Marvelous Mrs. Maisel - S02E08 - Someday. WEBDL-1080p.mkv failed

[23:59:38] scan: unrecognized file type

[23:59:38] libhb: scan thread found 0 valid title(s)

No title found.

HandBrake has exited.

Tdarr ALERT: NO OUTPUT FILE PRODUCEDC:/TransCache/Television/The Marvelous Mrs. Maisel - S02E08 - Someday… WEBDL-1080p-TdarrCacheFile-mXg-kWDoK.mp4
```

</p>
</details>  

**To Reproduce**
1. Set up server, webui and node on the machine hosting the media library
2. Share the transcode cache and media library directories
3. Map these shares to drives on the off-machine node
4. Add these paths to the node config
5. Change the node and server ip in the node config to their respective values
6. Spin up a transcode worker on the off-machine node

**Expected behavior**
The off-machine node should translate the paths correctly so that it can read and write to the transcode cache and media library.

**Screenshots**
The screenshot below shows that the mapped network drives can be accessed as expected
![image](https://user-images.githubusercontent.com/18357629/107441298-4b8e7c80-6b35-11eb-8042-c31efa53330e.png)

**Desktop (please complete the following information):**
Server:
 - OS: Windows Server 2019 v1809
 - Browser: Chrome v88.0.4324.150
 - Tdarr Version: 2.00.03

Node:
 - OS: Windows 10 v20H2
 - Browser: Chrome v88.0.4324.146
 - Tdarr Version: 2.00.03

**Additional context**
Steps already taken:
- Switching out escaped backslashes for forward slashes
- Run node as admin

**Comment by HaveAGitGat** [comment]
Hi are you able to please try with this sample file?

https://github.com/mediaelement/mediaelement-files/blob/master/big_buck_bunny.mp4

Your files may be copyright protected which HandBrake does not deal with.

**Comment by Fdebijl** [comment]
All my files are procured via usenet, so they shouldn't have any copyright protection. I also encountered the same error with Big Buck Bunny.

<details>
<summary>Node Logs</summary>
<p>

```
2.00.031
C:/Users/Parsec/Documents/Tdarr/configs/Tdarr_Node_Config.json
{
  nodeID: 'Parsec',
  nodeIP: '10.10.10.130',
  nodePort: '8267',
  serverIP: '10.10.10.6',
  serverPort: '8266',
  handbrakePath: '',
  ffmpegPath: '',
  mkvpropeditPath: '',
  pathTranslators: [
    { server: 'C:/TransCache/Television', node: 'Y:/Television' },
    { server: 'E:/Television', node: 'X:/Television' },
    { server: 'C:/TransCache/Debug', node: 'Y:/Debug' },
    { server: 'E:/Debug', node: 'X:/Debug' }
  ]
}
[2021-02-12T14:19:27.188] [INFO] Tdarr_Node - Starting Tdarr_Node
Preparing environment
[2021-02-12T14:19:27.293] [INFO] Tdarr_Node - Updating plugins
[2021-02-12T14:19:27.294] [INFO] Tdarr_Node - Cloning plugins
[2021-02-12T14:19:27.299] [INFO] Tdarr_Node - Tdarr_Node listening at http://localhost:8267
[2021-02-12T14:19:27.550] [INFO] Tdarr_Node - Finished downloading plugins!
[2021-02-12T14:19:28.740] [INFO] Tdarr_Node - ---------------Binary tests start----------------
[2021-02-12T14:19:28.743] [INFO] Tdarr_Node - handbrakePath:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/HandBrakeCLI.exe
[2021-02-12T14:19:28.746] [INFO] Tdarr_Node - ffmpegPath:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe
[2021-02-12T14:19:28.748] [INFO] Tdarr_Node - mkvpropedit:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/mkvpropedit/windows/mkvpropedit.exe
[2021-02-12T14:19:28.748] [INFO] Tdarr_Node - Binary test 1: handbrakePath working
[2021-02-12T14:19:28.749] [INFO] Tdarr_Node - Binary test 2: ffmpegPath working
[2021-02-12T14:19:28.749] [WARN] Tdarr_Node - Binary test 3: mkvpropeditPath not working:
[2021-02-12T14:19:28.750] [INFO] Tdarr_Node - ---------------Binary tests end-------------------
[2021-02-12T14:19:29.322] [INFO] Tdarr_Node - Node registered
[2021-02-12T14:19:30.084] [INFO] Tdarr_Node - [2.791s]Plugin update finished
[2021-02-12T14:19:42.990] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCEDC:/TransCache/Debug/big_buck_bunny-TdarrCacheFile-yOLABl4W2.mp4
```

</p>
</details>

**Comment by HaveAGitGat** [comment]
Seems the path translators are not working on your node.

In the Node log,
`C:/TransCache/Debug/big_buck_bun....`
Should be
`Y:/Debug/big_buck_bun....`

I've updated the path translating in 2.00.05 - still in pre-release so you need to set the version manually in the Tdarr Updater config:


```
{
  "platform_arch": "default",
  "modules": {
    "Tdarr_Desktop": {
      "enabled": true,
      "requiredVersion": "2.00.05",
      "currentVersion": ""
    },
    "Tdarr_Node": {
      "enabled": true,
      "requiredVersion": "2.00.05",
      "currentVersion": ""
    },
    "Tdarr_Server": {
      "enabled": true,
      "requiredVersion": "2.00.05",
      "currentVersion": ""
    },
    "Tdarr_WebUI": {
      "enabled": true,
      "requiredVersion": "2.00.05",
      "currentVersion": ""
    }
  }
}
```

Let me know if you still have the same issue, thanks.

**Comment by Fdebijl** [comment]
Thanks for the quick fix! There's no more path translation errors, but now the transcode doesn't seem to start at all:
![image](https://user-images.githubusercontent.com/18357629/107861504-80a51280-6e46-11eb-9d59-73bbd862b20e.png)

**Comment by HaveAGitGat** [comment]
Turn on verbose logs on the Logs tab and then post the node and server logs when that issue happens.

**Comment by Fdebijl** [comment]
I deleted the directories with the binaries and ran the updater again, which fixed the problem of the transcode not starting. Now the off-machine node still reports that no output file was produced, however.

<details>
<summary>Verbose Off-machine Node Logs</summary>
<p>

```
2.00.05
C:/Users/Parsec/Documents/Tdarr/configs/Tdarr_Node_Config.json
{
  nodeID: 'Parsec',
  nodeIP: '10.10.10.130',
  nodePort: '8267',
  serverIP: '10.10.10.6',
  serverPort: '8266',
  handbrakePath: '',
  ffmpegPath: '',
  mkvpropeditPath: '',
  pathTranslators: [
    { server: 'C:/TransCache/Television', node: 'Y:/Television' },
    { server: 'E:/Television', node: 'X:/Television' }
  ]
}
[2021-02-14T20:41:24.345] [INFO] Tdarr_Node - Starting Tdarr_Node
Preparing environment
[2021-02-14T20:41:24.358] [INFO] Tdarr_Node - Updating plugins
[2021-02-14T20:41:24.359] [INFO] Tdarr_Node - Cloning plugins
[2021-02-14T20:41:24.364] [INFO] Tdarr_Node - Tdarr_Node listening at http://localhost:8267
[2021-02-14T20:41:24.625] [INFO] Tdarr_Node - Finished downloading plugins!
[2021-02-14T20:41:25.757] [INFO] Tdarr_Node - ---------------Binary tests start----------------
[2021-02-14T20:41:25.760] [INFO] Tdarr_Node - handbrakePath:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/HandBrakeCLI.exe
[2021-02-14T20:41:25.762] [INFO] Tdarr_Node - ffmpegPath:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe
[2021-02-14T20:41:25.765] [INFO] Tdarr_Node - mkvpropedit:C:/Users/Parsec/Documents/Tdarr/Tdarr_Node/assets/app/mkvpropedit/windows/mkvpropedit.exe
[2021-02-14T20:41:25.766] [INFO] Tdarr_Node - Binary test 1: handbrakePath working
[2021-02-14T20:41:25.766] [INFO] Tdarr_Node - Binary test 2: ffmpegPath working
[2021-02-14T20:41:25.767] [WARN] Tdarr_Node - Binary test 3: mkvpropeditPath not working:
[2021-02-14T20:41:25.768] [INFO] Tdarr_Node - ---------------Binary tests end-------------------
[2021-02-14T20:41:26.909] [INFO] Tdarr_Node - Node registered
[2021-02-14T20:41:27.279] [INFO] Tdarr_Node - [2.921s]Plugin update finished
[2021-02-14T20:41:30.577] [INFO] Tdarr_Node - verbose:Launching worker
[2021-02-14T20:41:30.583] [INFO] Tdarr_Node - verbose:Worker kRBdI_Ieq launched
[2021-02-14T20:41:30.915] [INFO] Tdarr_Node - verbose:Requesting item
[2021-02-14T20:41:30.920] [INFO] Tdarr_Node - verbose:Item request received
[2021-02-14T20:41:30.920] [INFO] Tdarr_Node - verbose:Getting new task
[2021-02-14T20:41:30.932] [INFO] Tdarr_Node - verbose:Item received
{
  _id: "X:/Television/The Queen's Gambit/The Queen's Gambit - S01E03 - Doubled Pawns WEBDL-1080p.mkv",
  file: "X:/Television/The Queen's Gambit/The Queen's Gambit - S01E03 - Doubled Pawns WEBDL-1080p.mkv",
  container: 'mkv',
  ffProbeRead: 'success',
  createdAt: 1613331698579,
  lastPluginDetails: 'none',
  bit_rate: 6725688.120581689,
  statSync: {
    dev: 81867367,
    mode: 33206,
    nlink: 1,
    uid: 0,
    gid: 0,
    rdev: 0,
    blksize: 4096,
    ino: 10696049115055126,
    size: 2332818376,
    blocks: 4556288,
    atimeMs: 1612865026984.2822,
    mtimeMs: 1603496762813.348,
    ctimeMs: 1612865074399.5862,
    birthtimeMs: 1612864961001.831,
    atime: '2021-02-09T10:03:46.984Z',
    mtime: '2020-10-23T23:46:02.813Z',
    ctime: '2021-02-09T10:04:34.400Z',
    birthtime: '2021-02-09T10:02:41.002Z'
  },
  file_size: 2224.7489700317383,
  ffProbeData: {
    streams: [
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object]
    ]
  },
  meta: {
    SourceFile: "X:/Television/The Queen's Gambit/The Queen's Gambit - S01E03 - Doubled Pawns WEBDL-1080p.mkv",
    errors: [],
    Duration: 2774.816,
    PreviewDuration: 0,
    SelectionDuration: 0,
    TrackDuration: 2774.528,
    MediaDuration: 2701.625,
    ExifToolVersion: 12.1,
    FileName: "The Queen's Gambit - S01E03 - Doubled Pawns WEBDL-1080p.mkv",
    Directory: "X:/Television/The Queen's Gambit",
    FileSize: '2.2 GB',
    FileModifyDate: {
      year: 2020,
      month: 10,
      day: 24,
      hour: 1,
      minute: 46,
      second: 2,
      millisecond: 0,
      tzoffsetMinutes: 120,
      rawValue: '2020:10:24 01:46:02+02:00'
    },
    FileAccessDate: {
      year: 2021,
      month: 2,
      day: 9,
      hour: 11,
      minute: 3,
      second: 46,
      millisecond: 0,
      tzoffsetMinutes: 60,
      rawValue: '2021:02:09 11:03:46+01:00'
    },
    FileCreateDate: {
      year: 2021,
      month: 2,
      day: 9,
      hour: 11,
      minute: 2,
      second: 41,
      millisecond: 0,
      tzoffsetMinutes: 60,
      rawValue: '2021:02:09 11:02:41+01:00'
    },
    FilePermissions: 'rw-rw-rw-',
    FileType: 'MP4',
    FileTypeExtension: 'mp4',
    MIMEType: 'video/mp4',
    MajorBrand: 'MP4  Base Media v1 [IS0 14496-12:2003]',
    MinorVersion: '0.2.0',
    CompatibleBrands: [ 'isom', 'iso2', 'avc1', 'mp41' ],
    MediaDataSize: 2330884648,
    MediaDataOffset: 48,
    MovieHeaderVersion: 0,
    CreateDate: '0000:00:00 00:00:00',
    ModifyDate: '0000:00:00 00:00:00',
    TimeScale: 1000,
    PreferredRate: 1,
    PreferredVolume: '100.00%',
    PreviewTime: '0 s',
    PosterTime: '0 s',
    SelectionTime: '0 s',
    CurrentTime: '0 s',
    NextTrackID: 19,
    TrackHeaderVersion: 0,
    TrackCreateDate: '0000:00:00 00:00:00',
    TrackModifyDate: '0000:00:00 00:00:00',
    TrackID: 1,
    TrackLayer: 0,
    TrackVolume: '100.00%',
    Track1Name: 'allemand',
    Track2Name: 'anglais [VO]',
    Track3Name: 'espagnol',
    Track4Name: 'français',
    Track5Name: 'italien',
    Track6Name: 'allemand',
    Track7Name: 'anglais [VO] - Dolby',
    Track8Name: 'espagnol - Dolby',
    Track9Name: 'français',
    Track10Name: 'italien - Dolby',
    AudioFormat: 'mp4a',
    Track11Name: 'anglais - Description sonore',
    Balance: 0,
    AudioChannels: 2,
    AudioBitsPerSample: 16,
    AudioSampleRate: 48000,
    Track12Name: 'anglais - Description sonore - Dolby',
    ImageWidth: 1920,
    ImageHeight: 1080,
    GraphicsMode: 'srcCopy',
    OpColor: '0 0 0',
    CompressorID: 'avc1',
    SourceImageWidth: 1920,
    SourceImageHeight: 1080,
    XResolution: 72,
    YResolution: 72,
    BitDepth: 24,
    VideoFrameRate: 24,
    Track14Name: 'allemand',
    Track15Name: 'espagnol',
    Track16Name: 'français',
    Track17Name: 'italien',
    MatrixStructure: '1 0 0 0 1 0 0 0 1',
    MediaHeaderVersion: 0,
    MediaCreateDate: '0000:00:00 00:00:00',
    MediaModifyDate: '0000:00:00 00:00:00',
    MediaTimeScale: 1000,
    MediaLanguageCode: 'eng',
    HandlerDescription: 'SubtitleHandler',
    OtherFormat: 'tx3g',
    FontTable: 'Serif',
    Track18Name: 'anglais',
    HandlerType: 'Metadata',
    HandlerVendorID: 'Apple',
    Encoder: 'Lavf58.18.101',
    ImageSize: '1920x1080',
    Megapixels: 2.1,
    AvgBitrate: '6.72 Mbps',
    Rotation: 0
  },
  mediaInfo: {
    '@ref': '',
    track: [
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object]
    ]
  },
  hasClosedCaptions: false,
  DB: 'MiIYQMvps',
  bumped: false,
  HealthCheck: '',
  TranscodeDecisionMaker: '',
  processingStatus: false,
  fileMedium: 'video',
  infoLog: '',
  video_codec_name: 'aac',
  audio_codec_name: '',
  video_resolution: 'Other',
  lastHealthCheckDate: 0,
  lastTranscodeDate: 0,
  history: '',
  oldSize: 0,
  newSize: 0,
  tPosition: 0,
  hPosition: 0
}
{
  processFile: true,
  preset: ',-map 0:v? -map 0:a? -map 0:s? -map 0:d? -map 0:t? -c copy',
  container: '.mkv',
  handBrakeMode: false,
  FFmpegMode: true,
  reQueueAfter: true,
  infoLog: 'Video is not in the first stream',
  lastPluginDetails: {
    source: 'Community',
    id: 'Tdarr_Plugin_lmg1_Reorder_Streams',
    number: '1/6'
  },
  ffmpegMode: true
}
[2021-02-14T20:41:40.840] [INFO] Tdarr_Node - verbose:Sub-worker exit status received
[2021-02-14T20:41:40.841] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCED

C:/TransCache/Television/The Queen's Gambit - S01E03 - Doubled Pawns WEBDL-1080p-TdarrCacheFile-Nvugriyn5.mkv
[2021-02-14T20:41:40.847] [INFO] Tdarr_Node - verbose:Requesting item
[2021-02-14T20:41:40.848] [INFO] Tdarr_Node - verbose:Item request received
[2021-02-14T20:41:40.848] [INFO] Tdarr_Node - verbose:Getting new task
[2021-02-14T20:41:40.859] [INFO] Tdarr_Node - verbose:Item received
{
  _id: "X:/Television/The Queen's Gambit/The Queen's Gambit - S01E02 - Exchanges WEBDL-1080p.mkv",
  file: "X:/Television/The Queen's Gambit/The Queen's Gambit - S01E02 - Exchanges WEBDL-1080p.mkv",
  container: 'mkv',
  ffProbeRead: 'success',
  createdAt: 1613331707225,
  lastPluginDetails: 'none',
  bit_rate: 6370770.403781825,
  statSync: {
    dev: 81867367,
    mode: 33206,
    nlink: 1,
    uid: 0,
    gid: 0,
    rdev: 0,
    blksize: 4096,
    ino: 20266198323212596,
    size: 3115790906,
    blocks: 6085536,
    atimeMs: 1612864954012.9802,
    mtimeMs: 1603508646000,
    ctimeMs: 1612864976695.0703,
    birthtimeMs: 1612864878307.3347,
    atime: '2021-02-09T10:02:34.013Z',
    mtime: '2020-10-24T03:04:06.000Z',
    ctime: '2021-02-09T10:02:56.695Z',
    birthtime: '2021-02-09T10:01:18.307Z'
  },
  file_size: 2971.4497623443604,
  ffProbeData: {
    streams: [
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object]
    ]
  },
  meta: {
    SourceFile: "X:/Television/The Queen's Gambit/The Queen's Gambit - S01E02 - Exchanges WEBDL-1080p.mkv",
    errors: [],
    Duration: 3912.608,
    PreviewDuration: 0,
    SelectionDuration: 0,
    TrackDuration: 3912.576,
    MediaDuration: 3740.83,
    ExifToolVersion: 12.1,
    FileName: "The Queen's Gambit - S01E02 - Exchanges WEBDL-1080p.mkv",
    Directory: "X:/Television/The Queen's Gambit",
    FileSize: '2.9 GB',
    FileModifyDate: {
      year: 2020,
      month: 10,
      day: 24,
      hour: 5,
      minute: 4,
      second: 6,
      millisecond: 0,
      tzoffsetMinutes: 120,
      rawValue: '2020:10:24 05:04:06+02:00'
    },
    FileAccessDate: {
      year: 2021,
      month: 2,
      day: 9,
      hour: 11,
      minute: 2,
      second: 34,
      millisecond: 0,
      tzoffsetMinutes: 60,
      rawValue: '2021:02:09 11:02:34+01:00'
    },
    FileCreateDate: {
      year: 2021,
      month: 2,
      day: 9,
      hour: 11,
      minute: 1,
      second: 18,
      millisecond: 0,
      tzoffsetMinutes: 60,
      rawValue: '2021:02:09 11:01:18+01:00'
    },
    FilePermissions: 'rw-rw-rw-',
    FileType: 'MP4',
    FileTypeExtension: 'mp4',
    MIMEType: 'video/mp4',
    MajorBrand: 'MP4  Base Media v1 [IS0 14496-12:2003]',
    MinorVersion: '0.2.0',
    CompatibleBrands: [ 'isom', 'iso2', 'avc1', 'mp41' ],
    MediaDataSize: 3113099566,
    MediaDataOffset: 48,
    MovieHeaderVersion: 0,
    CreateDate: '0000:00:00 00:00:00',
    ModifyDate: '0000:00:00 00:00:00',
    TimeScale: 1000,
    PreferredRate: 1,
    PreferredVolume: '100.00%',
    PreviewTime: '0 s',
    PosterTime: '0 s',
    SelectionTime: '0 s',
    CurrentTime: '0 s',
    NextTrackID: 19,
    TrackHeaderVersion: 0,
    TrackCreateDate: '0000:00:00 00:00:00',
    TrackModifyDate: '0000:00:00 00:00:00',
    TrackID: 1,
    TrackLayer: 0,
    TrackVolume: '100.00%',
    Track1Name: 'allemand',
    Track2Name: 'anglais [VO]',
    Track3Name: 'espagnol',
    Track4Name: 'français',
    Track5Name: 'italien',
    Track6Name: 'allemand - Dolby',
    Track7Name: 'anglais [VO]',
    Track8Name: 'espagnol - Dolby',
    Track9Name: 'français',
    Track10Name: 'italien - Dolby',
    AudioFormat: 'mp4a',
    Track11Name: 'anglais - Description sonore',
    Balance: 0,
    AudioChannels: 2,
    AudioBitsPerSample: 16,
    AudioSampleRate: 48000,
    Track12Name: 'anglais - Description sonore - Dolby',
    ImageWidth: 1920,
    ImageHeight: 1080,
    GraphicsMode: 'srcCopy',
    OpColor: '0 0 0',
    CompressorID: 'avc1',
    SourceImageWidth: 1920,
    SourceImageHeight: 1080,
    XResolution: 72,
    YResolution: 72,
    BitDepth: 24,
    VideoFrameRate: 24,
    Track14Name: 'allemand',
    Track15Name: 'espagnol',
    Track16Name: 'français',
    Track17Name: 'italien',
    MatrixStructure: '1 0 0 0 1 0 0 0 1',
    MediaHeaderVersion: 0,
    MediaCreateDate: '0000:00:00 00:00:00',
    MediaModifyDate: '0000:00:00 00:00:00',
    MediaTimeScale: 1000,
    MediaLanguageCode: 'eng',
    HandlerDescription: 'SubtitleHandler',
    OtherFormat: 'tx3g',
    FontTable: 'Serif',
    Track18Name: 'anglais',
    HandlerType: 'Metadata',
    HandlerVendorID: 'Apple',
    Encoder: 'Lavf58.18.101',
    ImageSize: '1920x1080',
    Megapixels: 2.1,
    AvgBitrate: '6.37 Mbps',
    Rotation: 0
  },
  mediaInfo: {
    '@ref': '',
    track: [
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object], [Object], [Object],
      [Object]
    ]
  },
  hasClosedCaptions: false,
  DB: 'MiIYQMvps',
  bumped: false,
  HealthCheck: '',
  TranscodeDecisionMaker: '',
  processingStatus: false,
  fileMedium: 'video',
  infoLog: '',
  video_codec_name: 'aac',
  audio_codec_name: '',
  video_resolution: 'Other',
  lastHealthCheckDate: 0,
  lastTranscodeDate: 0,
  history: '',
  oldSize: 0,
  newSize: 0,
  tPosition: 0,
  hPosition: 0
}
{
  processFile: true,
  preset: ',-map 0:v? -map 0:a? -map 0:s? -map 0:d? -map 0:t? -c copy',
  container: '.mkv',
  handBrakeMode: false,
  FFmpegMode: true,
  reQueueAfter: true,
  infoLog: 'Video is not in the first stream',
  lastPluginDetails: {
    source: 'Community',
    id: 'Tdarr_Plugin_lmg1_Reorder_Streams',
    number: '1/6'
  },
  ffmpegMode: true
}
[2021-02-14T20:41:49.407] [INFO] Tdarr_Node - verbose:Sub-worker exit status received
[2021-02-14T20:41:49.408] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCED

C:/TransCache/Television/The Queen's Gambit - S01E02 - Exchanges WEBDL-1080p-TdarrCacheFile-D0r_ABIX0.mkv
```

</p>
</details>  

The file properties in the verbose object dump seem to indicate the path translation is being applied properly to the media directory, but not to the transcode cache. I should probably note I'm rolling my own plugins. Those worked fine on v1, but I'm not sure if I should be making any changes for v2.

**Comment by HaveAGitGat** [comment]
I have not seen this issue before. Perhaps you have a hidden character in the path translators Node config json file? You can check for hidden characters here:
https://www.soscisurvey.de/tools/view-chars.php

Perhaps try typing out the paths manually or paste your Tdarr_Node_Config.json file here

**Comment by Fdebijl** [comment]
I checked for hidden characters, but there's nothing but the usual carriage returns and line breaks.

(Off-machine) Node Config
```
{
  "nodeID": "Parsec",
  "nodeIP": "10.10.10.130",
  "nodePort": "8267",
  "serverIP": "10.10.10.6",
  "serverPort": "8266",
  "handbrakePath": "",
  "ffmpegPath": "",
  "mkvpropeditPath": "",
  "pathTranslators": [
    {
      "server": "C:/TransCache/Television",
      "node": "Y:/Television"
    },
    {
      "server": "E:/Television",
      "node": "X:/Television"
    }
  ]
}
```

Server and node config from the master machine
```{
  "serverIP": "0.0.0.0",
  "serverPort": "8266",
  "handbrakePath": "",
  "ffmpegPath": ""
}
```

```
{
  "nodeID": "Daphnis",
  "nodeIP": "127.0.0.1",
  "nodePort": "8267",
  "serverIP": "127.0.0.1",
  "serverPort": "8266",
  "handbrakePath": "",
  "ffmpegPath": "",
  "mkvpropeditPath": "",
  "pathTranslators": [
    {
      "server": "",
      "node": ""
    }
  ]
}
```

**Comment by HaveAGitGat** [comment]
I think I may have found the issue.

Am I right in thinking you are using back slashes in the transcode cache path in the library settings?

Try using forward slashes:
![image](https://user-images.githubusercontent.com/43864057/108169407-f9cb9080-70f8-11eb-9434-c10fb92441e7.png)

I've fixed that for the next release. This was to do with the change allowing Windows UNC paths a few weeks ago.

**Comment by Scaredycrow** [comment]
Chiming in:  I have an almost identical issue when using a windows node connecting to the server (docker container) on a linux host. Also happen to be running 2.00.05. 

From reddit thread here: https://www.reddit.com/r/Tdarr/comments/lkw7ly/unable_to_get_a_win10_node_transcoding_with_a/ 

Windows Node Log: 

```
[2021-02-16T15:44:22.583] [INFO] Tdarr_Node - Starting Tdarr_Node
Preparing environment
[2021-02-16T15:44:22.839] [INFO] Tdarr_Node - Path translator: Checking Node path: z:\tdarr_transcode_cache
[2021-02-16T15:44:22.937] [INFO] Tdarr_Node - Updating plugins
[2021-02-16T15:44:22.938] [INFO] Tdarr_Node - Cloning plugins
[2021-02-16T15:44:22.943] [INFO] Tdarr_Node - Tdarr_Node listening at http://localhost:8267
[2021-02-16T15:44:22.949] [INFO] Tdarr_Node - Path translator: Success: Node path can be accessed: z:\tdarr_transcode_cache
[2021-02-16T15:44:22.949] [INFO] Tdarr_Node - Path translator: Checking Node path: z:\tv_shows
[2021-02-16T15:44:22.952] [INFO] Tdarr_Node - Path translator: Success: Node path can be accessed: z:\tv_shows
[2021-02-16T15:44:23.731] [INFO] Tdarr_Node - ---------------Binary tests start----------------
[2021-02-16T15:44:23.734] [INFO] Tdarr_Node - handbrakePath:C:/Temp/Tdarr_Updater/Tdarr_Node/assets/app/HandBrakeCLI.exe[2021-02-16T15:44:23.736] [INFO] Tdarr_Node - ffmpegPath:C:/Temp/Tdarr_Updater/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe
[2021-02-16T15:44:23.738] [INFO] Tdarr_Node - mkvpropedit:C:/Temp/Tdarr_Updater/Tdarr_Node/assets/app/mkvpropedit/windows/mkvpropedit.exe
[2021-02-16T15:44:23.738] [INFO] Tdarr_Node - Binary test 1: handbrakePath working
[2021-02-16T15:44:23.739] [INFO] Tdarr_Node - Binary test 2: ffmpegPath working
[2021-02-16T15:44:23.739] [WARN] Tdarr_Node - Binary test 3: mkvpropeditPath not working:
[2021-02-16T15:44:23.739] [INFO] Tdarr_Node - ---------------Binary tests end-------------------
[2021-02-16T15:44:23.848] [INFO] Tdarr_Node - Finished downloading plugins!
[2021-02-16T15:44:24.972] [INFO] Tdarr_Node - Node registered
[2021-02-16T15:44:26.304] [INFO] Tdarr_Node - [3.366s]Plugin update finished
The following external module needs to be installed for a plugin:import-fresh
npm WARN saveError ENOENT: no such file or directory, open 'C:\Temp\Tdarr_Updater\Tdarr_Node\assets\app\plugins\package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Temp\Tdarr_Updater\Tdarr_Node\assets\app\plugins\package.json'
npm WARN plugins No description
npm WARN plugins No repository field.
npm WARN plugins No README data
npm WARN plugins No license field.

+ import-fresh@3.3.0
added 4 packages from 1 contributor and audited 4 packages in 0.708s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

[2021-02-16T15:44:52.733] [WARN] Tdarr_Node - Tdarr ALERT: NO OUTPUT FILE PRODUCED:
z:/tdarr_transcode_cache/All or Nothing - New Zealand All Blacks - S01E06 - Your Greatest Rival is Yourself-TdarrCacheFile-bgoY7pcAy.mkv
```

Output of that file from the webui: 

```
Command:
C:/Temp/Tdarr_Updater/Tdarr_Node/node_modules/@ffmpeg-installer/win32-x64/ffmpeg.exe -i /mediaz:/tv_shows_shows/All or Nothing- New Zealand All Blacks/Season 01/All or Nothing - New Zealand All Blacks - S01E06 - Your Greatest Rival is Yourself.mkv -map 0 -c:v copy -c:a copy -map 0:1 -c:a:0 aac -ac 2 -metadata:s:a:0 title="2.0 " -strict -2 -c:s copy -max_muxing_queue_size 9999 z:/tdarr_transcode_cache/All or Nothing - New Zealand All Blacks - S01E06 - Your Greatest Rival is Yourself-TdarrCacheFile-bgoY7pcAy.mkv
Last 200 lines of CLI log:
ffmpeg version N-92722-gf22fcd4483 Copyright (c) 2000-2018 the FFmpeg developers
built with gcc 8.2.1 (GCC) 20181201
configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-amf --enable-ffnvcodec --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt
libavutil 56. 24.101 / 56. 24.101
libavcodec 58. 42.102 / 58. 42.102
libavformat 58. 24.101 / 58. 24.101
libavdevice 58. 6.101 / 58. 6.101
libavfilter 7. 46.101 / 7. 46.101
libswscale 5. 4.100 / 5. 4.100
libswresample 3. 4.100 / 3. 4.100
libpostproc 55. 4.100 / 55. 4.100
/mediaz:/tv_shows_shows/All or Nothing- New Zealand All Blacks/Season 01/All or Nothing - New Zealand All Blacks - S01E06 - Your Greatest Rival is Yourself.mkv: Invalid argument
Tdarr ALERT: NO OUTPUT FILE PRODUCED:
z:/tdarr_transcode_cache/All or Nothing - New Zealand All Blacks - S01E06 - Your Greatest Rival is Yourself-TdarrCacheFile-bgoY7pcAy.mkv
```

The odd / interesting part here is the line beginning: 

"/mediaz:/tv_shows_shows/"

That's just not even close to the node config file (below), and appears to be some conflation of the server side docker mapping (/media/tv_shows) and the windows node config (z:\tv_shows). 

Is this related to this bug report, or should I be opening a new one? 

Windows node config: 

```
{
  "nodeID": "host-name",
  "nodeIP": "192.168.1.130",
  "nodePort": "8267",
  "serverIP": "192.168.1.6",
  "serverPort": "8266",
  "handbrakePath": "",
  "ffmpegPath": "",
  "mkvpropeditPath": "",
  "pathTranslators": [
    {
      "server": "/temp",
      "node": "z:\\tdarr_transcode_cache"
    },
    {
      "server": "/tv",
      "node": "z:\\tv_shows"
    }
  ]
}
```

**Comment by Fdebijl** [comment]
> I think I may have found the issue.
> 
> Am I right in thinking you are using back slashes in the transcode cache path in the library settings? 
> ...

You are absolutely correct, switching the backslashes for slashes resolved my issue! Thank you so much for your time and effort.

This may also extend to @Scaredycrow, it might work as expected using slashes rather than escaped backslashes in the node config.

**Comment by Scaredycrow** [comment]
> > I think I may have found the issue.
> > Am I right in thinking you are using back slashes in the transcode cache path in the library settings?
> > ...
> 
> You are absolutely correct, switching the backslashes for slashes resolved my issue! Thank you so much for your time and effort.
> 
> This may also extend to @Scaredycrow, it might work as expected using slashes rather than escaped backslashes in the node config.

Note: This did not fix my issue, although forward slashes do now seem to work when mapping paths, the error on my end remains the same.

**Comment by HaveAGitGat** [comment]
> > I think I may have found the issue.
> > Am I right in thinking you are using back slashes in the transcode cache path in the library settings?
> > ...
> 
> You are absolutely correct, switching the backslashes for slashes resolved my issue! Thank you so much for your time and effort.
> 
> This may also extend to @Scaredycrow, it might work as expected using slashes rather than escaped backslashes in the node config.

Great :) Thanks for the help.

**Comment by HaveAGitGat** [comment]
> > > I think I may have found the issue.
> > > Am I right in thinking you are using back slashes in the transcode cache path in the library settings?
> > > ...
> > 
> > 
> > You are absolutely correct, switching the backslashes for slashes resolved my issue! Thank you so much for your time and effort.
> > This may also extend to @Scaredycrow, it might work as expected using slashes rather than escaped backslashes in the node config.
> 
> Note: This did not fix my issue, although forward slashes do now seem to work when mapping paths, the error on my end remains the same.

Yours is a different issue. Please open a new issue if you're still having that problem, thanks.

**Comment by janky5ive** [comment]
@Scaredycrow I have the same issue and have created a new issue (I didnt see you create one after your comments in this thread). Issue: https://github.com/HaveAGitGat/Tdarr/issues/1132
